### ISSUE TITLE
Add " Keep searching when switching 'tabs' " option

### ISSUE THREAD

**Issue by a-pav** [mbnuqw/sidebery#1527]
`Keep searching when switching 'tabs'`

Currently switching to any tab after a search is made, clears the search. This way the user can't move around between the filtered tabs without losing what s/he has searched for.

**Comment by devgid** [comment]
I have the exact same request. The moment after a search when you open a tab, it clears the search filter and shows all tabs unfiltered.
Some might prefer that behavior but it would be nice if there is an option in settings like @a-pav request to not clear the search filter and not to unfilter the tabs when you click on a tab after a search.

**Comment by rieje** [comment]
Agreed, filtering tabs is such a useful feature but if dealing with even more than one filtered tab result it's annoying to do another filtered search. There's already the button to clear the filtered search and return to tabs list--an option to preserve filtered tabs list on selected filtered tab is essential IMO.

**Comment by mbrasch** [comment]
I really miss that too.

**Comment by hktonylee** [comment]
I'd love to see this feature implemented =)

**Comment by mbrasch** [comment]
THANK YOU! You have made a simple man very happy. :)

**Comment by devgid** [comment]
I have tested this in latest version and when switching tabs, the search is reset and all tabs are shown again in the sidebar.

Couldn't find any search option in the settings that would allow the new behavior.

In addition now when closing an existing tab (press x close in the sidebar without switching) it now moves the whole sidebar to the top which didn't happen in the previous version

**Comment by silverwings15** [comment]
> Couldn't find any search option in the settings that would allow the new behavior.

<img width="1102" height="433" alt="Image" src="https://github.com/user-attachments/assets/d5bc1e0d-4c2b-494e-98b2-2693224488d7" />

**Comment by devgid** [comment]
> > Couldn't find any search option in the settings that would allow the new behavior.
> 
> <img alt="Image" width="1102" height="433" src="https://private-user-images.githubusercontent.com/10742312/554320735-d5bc1e0d-4c2b-494e-98b2-2693224488d7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzE5NTc1NjUsIm5iZiI6MTc3MTk1NzI2NSwicGF0aCI6Ii8xMDc0MjMxMi81NTQzMjA3MzUtZDViYzFlMGQtNGMyYi00OTRlLTk4YjItMjY5MzIyNDQ4OGQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjI0VDE4MjEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkZTQ2YWI0NGVmMTU4NjBmMDA1ZmZkNzEwZjE1MDQwZjc4OTdmMGU1ZTFmNzQxNmIzOTQyYjM2MmQyNWI3MTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nXf6_zY_TJYSA9albgDa4kvceGSWlGR9AdoDdHe5ze4">

Missed it. Thanks. Description would be clearer if read. "Keep search filter after switching tabs"

**Comment by devgid** [comment]
Only issue with search for me now is when clicking on tab in sidebar to open tab when Search filter has text, then the sidebar view jumps to the top, and the active tab in the sidebar is lost.

Same applies when NOT navigating to new tab but just closing a tab the sidebar jumps to the top if there is search filter . Even with the "Keep Searching" setting off, the sidebar view jumps to the top of the sidebar. This is something the Nov 2025 version did not do so it seems to be something that got broken.

**Comment by silverwings15** [comment]
i think i know the bug you're talking about, please try to see if you can reproduce: the active tab _indicator_ disappears when, while search is active, to open a new tab _with Ctrl+T_. middle clicking links does not trigger this bug.

edit: upon reading your edit, this probably is not the same bug as yours

on that note, does anyone get this other bug? when pressing Ctrl+T to open new tabs, all the displayed tabs temporarily disappear while this message shows up for half a second to one second. been experiencing it since the release of the new version, although i haven't reported it because i haven't attempted to reproduce it on a fresh setup

<img width="359" height="561" alt="Image" src="https://github.com/user-attachments/assets/6d518c92-6148-43ee-9b79-6db55fd67a72" />

**Comment by devgid** [comment]
> i think i know the bug you're talking about, please try to see if you can reproduce: the active tab _indicator_ disappears when, while search is active, to open a new tab _with Ctrl+T_. middle clicking links does not trigger this bug.

That behavior is logical because a new tab is not being filtered by search and it navigates away from the tab. 

Also Ctrl+ T would be outside the scope of Sidebery just like Ctrl+W, instead of clicking on the x next to the tab will result in not going to the next filtered tab but to a non visible tab because Ctrl+W  is a browser shortcut, not a Sidebery shortcut.

**Comment by mbnuqw** [comment]
@devgid,
> Only issue with search for me now is...

Thanks for the report, can repro, will fix.

@silverwings15,
> when pressing Ctrl+T to open new tabs, all the displayed tabs temporarily disappear

This happens when Sidebery detects that native tabs and Sidebery's tabs are unexpectedly desynchronized, which triggers tabs reinitialization. Previously, this was done silently, relatively fast, but required complicated logic (read, bug-prone) and might lead to tree structure collapse, accumulating issues with tree structure and so on... (there are tons of reports). With v5.5.0, I replaced old reinit logic with full unload/load cycle. It takes more time but is much more reliable and also shows that "Processing tabs..." banner, which might motivate user to report this issue and help me to find the cause, fix/add workaround.

Now, about the possible cause. There are three options: a sidebery bug with tabs, interference with other addon and/or some non-default about:config flag that I wasn't aware of. The fact, that you're unable to reproduce this on a fresh profile means that it's probably second or third option. Could you try to find possibly interfering addons by disabling them one by one, recheck about:config flags and then open a new bug report with that info (+ with steps to reproduce on your current profile)? When I got more time, I'll check it out.

**Comment by silverwings15** [comment]
> That behavior is logical because a new tab is not being filtered by search and it navigates away from the tab.

no, i mean that even when you return from the newly opened tab to one of the filtered tabs, the selected filtered doesn't show any indication that it is the active tab. still working on reliably reproducing this though, it's not happened much on the new profile i'm testing

> Could you try to find possibly interfering addons by disabling them one by one, recheck about:config flags and then open a new bug report with that info

so far it doesn't look like about:config flags are the culprit. i'll keep digging when i have the time

**Comment by silverwings15** [comment]
i found the culprit to be this addon: https://github.com/sblask/webextension-open-tabs-next-to-current

**Comment by llc0930** [comment]
> <img width="1102" height="433" alt="Image" src="https://github.com/user-attachments/assets/1de17ad3-9832-4d7c-bc19-562c8b092531" />

My search settings match the image above.

If it helps, the only add-ons I have left are the ones shown below.

<img width="285" height="596" alt="Image" src="https://github.com/user-attachments/assets/f55376d4-2331-460e-9854-e0ae9fe707da" />

I’ve noticed the search keeps looping between success and failure, just like in the video.

https://github.com/user-attachments/assets/33720dbd-ab5f-4de7-b6b3-ec738c427ac5
